### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by me-liu**
### Description

[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- moved everything over to a new branch, so this PR should be the same as the other one
- all functionality should work 

### Screenshots

[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

## How to review

[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- the commits aren't very meaningful as this branch was copy pasted

## Next steps

[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- need to add in SVGs 
- need the button t o change from gray to black when everything is entered correctly

### Online sources

[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @stephaniewong2

### CODE REVIEW

File: src/app/(Authentication)/Login/Email/index.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  return (
+    <View style={styles.container}>
+      <TouchableOpacity style={styles.backButton} onPress={() => router.back()}>
+        <Text style={styles.backText}>Back</Text>
+      </TouchableOpacity>
+      <Text style={styles.instructionText}>
+        Please enter your email address.
+      </Text>
+      <Text style={styles.emailText}>
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Email/index.tsx]: Let's make this `Text` and `TextInput` a styled component because we use it across multiple screens. I can show you more and explain this Wednesday!

File: src/app/(Authentication)/Login/Email/index.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+export default function LoginScreen() {
+  const [email, setEmail] = useState('');
+  const [isEmail, setIsEmail] = useState(false);
+  const [displayError, setDisplayError] = useState(false);
+  const [displayEmail, setDisplayEmail] = useState(false);
+  const [placeholder, setPlaceholder] = useState('Email');
+  const [isFocused, setIsFocused] = useState(false);
+
+  const onClick = () => {
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Email/index.tsx]: We can move all these functions into the styled component as well

File: src/app/(Authentication)/Login/Password/index.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import { router, useLocalSearchParams } from 'expo-router';
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Password/index.tsx:1]: Same comments as Email screen

File: src/app/(Authentication)/Login/Password/styles.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    height: 50,
+    padding: 15,
+    opacity: 0.3,
+  },
+  nextButton: {
+    backgroundColor: colors.white,
+    alignItems: 'center',
+    marginBottom: 20,
+    borderRadius: 3,
+    borderColor: '#000000',
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Password/styles.js]: Let's use the colors file

File: src/app/(Authentication)/Login/Email/styles.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    borderRadius: 3,
+    borderWidth: 1,
+    borderColor: colors.black,
+    width: 316,
+    height: 50,
+    padding: 15,
+    opacity: 0.35,
+  },
+  nextButton: {
+    backgroundColor: '#fff',
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Email/styles.js]: Let's use the colors file

File: src/app/(Authentication)/Login/Email/styles.js
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Email/styles.js:1]: These style sheets should be typescript files not javascript!

File: src/app/(Authentication)/Login/Password/index.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    }
+  }
+
+  async function signIn() {
+    const isPassword = await passwordExists(email, password);
+    if (!isPassword) {
+      setDisplayError(true);
+    } else {
+      setDisplayError(false);
+      signInUser(email, password);
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Password/index.tsx]: We don't need to call `signInUser` here because it already signs in the user if the correct credentials are given through `passwordExists`.

File: src/app/(Authentication)/Login/Password/index.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+      />
+
+      <Text style={styles.errorMessage}>
+        {displayError
+          ? 'Oh no! The password you entered is incorrect, please try again.'
+          : ' '}{' '}
+      </Text>
+
+      <View>
+        <TouchableOpacity style={[styles.nextButton]} onPress={() => signIn()}>
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Password/index.tsx]: Remove square brackets from styling

File: src/app/(Authentication)/Login/Email/index.tsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        autoCapitalize="none"
+        clearButtonMode="while-editing"
+      />
+      <Text style={styles.errorMessage}>
+        {displayError
+          ? 'The email you entered is either incorrect or not registered with the Impact Fund.'
+          : ' '}
+      </Text>
+      <View>
+        <TouchableOpacity style={[styles.nextButton]} onPress={emailFind}>
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Email/index.tsx]: Remove square brackets from styling

File: src/app/(Authentication)/Login/Email/index.tsx (hunk 1 above)
Comment (stephaniewong2) [reply src/app/(Authentication)/Login/Email/index.tsx]: We can essentially make a `AuthInput` component that takes in the necessary state variables and other props that does the same thing as this but more generally so it can be used across different screens.

**Review by stephaniewong2 — CHANGES_REQUESTED** [review]
Looks so good! Let's make a styled `AuthInput` so we can use it across the app!

File: src/Components/AuthInput/AuthInput.tsx [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+interface AuthInputProps {
+  input: string;
+  setInput: React.Dispatch<React.SetStateAction<string>>;
+  defaultValue: string;
+  isPassword: boolean;
+  displayInput: boolean;
+  setDisplayInput: React.Dispatch<React.SetStateAction<boolean>>;
+  keyboard: KeyboardTypeOptions;
+  autoCap: boolean;
```
Comment (stephaniewong2) [inline src/Components/AuthInput/AuthInput.tsx]: Nit but can we call this `autoCapitalization` instead of `autoCap`

File: src/Components/AuthInput/styles.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    borderBottomWidth: 0.75,
+    width: 310,
+    borderColor: colors.darkGrey,
+  },
+  inputFocused: {
+    height: 40,
+    borderBottomWidth: 2,
+    width: 310,
+    borderColor: colors.darkRed,
+    borderRadius: 2,
```
Comment (stephaniewong2) [inline src/Components/AuthInput/styles.ts]: Remove `borderRadius`

File: src/Components/AuthInput/AuthInput.tsx [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+  input,
+  setInput,
+  defaultValue,
+  isPassword,
+  displayInput,
+  setDisplayInput,
+  keyboard,
+  autoCap,
+}: AuthInputProps) {
+  const [placeholder, setPlaceholder] = useState<string>(defaultValue);
```
Comment (stephaniewong2) [inline src/Components/AuthInput/AuthInput.tsx]: Remove this and instead pass down `placeholder` and `setPlaceholder` as props. This will fix the bug

File: src/app/(Authentication)/Login/Email/index.tsx [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+import { Text, TouchableOpacity, View } from 'react-native';
+
+import styles from './styles';
+import AuthInput from '../../../../Components/AuthInput/AuthInput';
+import { emailExists } from '../../../../supabase/queries/auth';
+
+export default function LoginScreen() {
+  const [email, setEmail] = useState('');
+  const [displayError, setDisplayError] = useState(false);
+  const [displayEmail, setDisplayEmail] = useState(false);
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Email/index.tsx]: Add a `placeholder` `useState`, add this to the `AuthInputProps` in the `AuthInput` component, and pass down the variable and its function.

File: src/app/(Authentication)/Login/Email/index.tsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+  async function emailFind() {
+    const isEmail = await emailExists(email);
+    if (!isEmail) {
+      setDisplayError(true);
+    } else {
+      setDisplayError(false);
+      router.push({ pathname: 'Login/Password', params: { email } });
+      setEmail('');
+      setDisplayEmail(false);
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Email/index.tsx:25]: Then here we can call `setPlaceholder('Email address')`

File: src/app/(Authentication)/Login/Email/index.tsx [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+import React, { useState } from 'react';
+import { Text, TouchableOpacity, View } from 'react-native';
+
+import styles from './styles';
+import AuthInput from '../../../../Components/AuthInput/AuthInput';
+import { emailExists } from '../../../../supabase/queries/auth';
+
+export default function LoginScreen() {
+  const [email, setEmail] = useState('');
+  const [displayError, setDisplayError] = useState(false);
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Email/index.tsx]: Type all `useStates`

File: src/app/(Authentication)/Login/Password/index.tsx [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+
+import styles from './styles';
+import AuthInput from '../../../../Components/AuthInput/AuthInput';
+import { passwordExists } from '../../../../supabase/queries/auth';
+
+export default function LoginScreen() {
+  const [password, setPassword] = useState('');
+  const [displayError, setDisplayError] = useState(false);
+  const { email } = useLocalSearchParams() as unknown as { email: string };
+  const [displayPassword, setDisplayPassword] = useState(false);
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Password/index.tsx]: We can do the same thing here. Add a `placeholder` state variable and pass it down to `AuthInput`

File: src/app/(Authentication)/Login/Password/index.tsx [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+
+  async function signIn() {
+    const isPassword = await passwordExists(email, password);
+    if (!isPassword) {
+      setDisplayError(true);
+    } else {
+      setDisplayError(false);
+      setPassword('');
```
Comment (stephaniewong2) [inline src/app/(Authentication)/Login/Password/index.tsx:22]: Then here set the placeholder value to Password

**Review by stephaniewong2 — CHANGES_REQUESTED** [review]
Looks good! Just left comments for resolving bug!
